File: program205.cpp
#include<iostream>
using namespace std;

int SumFactors(int iNo)
{
    int iSum = 0, iCnt = 0;

    for(iCnt = 1; iCnt < iNo/2; iCnt++)
    {
        if((iNo % iCnt) == 0)
        {
            iSum = iCnt + iSum;
        }
    }
    return iSum;
}
int main()
{
    int iValue = 0;
    int iRet = 0;

    cout<<"Enter first number : "<<endl;
    cin>>iValue;

    iRet = SumFactors(iValue);

    cout<<"Sum of factors is : "<<iRet<<endl;

    return 0;
}

File: program205.java
import java.util.*;

class program205
{
    public static void main(String Args[])
    {
        int iValue = 0, iCnt = 0, iSum = 0;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the elements : ");
        iValue = sobj.nextInt();

        for(iCnt = 1; iCnt <= (iValue / 2); iCnt++)
        {
            if((iValue % iCnt) == 0)
            {
                iSum = iSum + iCnt;
            }
        }
        System.out.println("Sum of factors is : "+iSum);
    }
}


